src/ws/cockpitchannelresponse.c

```diff
@@ -76,17 +76,29 @@ cockpit_channel_inject_perform (CockpitChannelInject *inject,
     return;
 
   creds = cockpit_web_service_get_creds (inject->service);
-  application = cockpit_creds_get_application (creds);
+  if (cockpit_web_response_get_url_root (response))
```

Comment:
Why is the URL root a per response thing? The configurability of this gives me the impression that it's a global option.